libs/dataset-explorer/src/lib/DefaultTab.tsx

```diff
@@ -0,0 +1,389 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  AxisConfigDialog,
+  JointDataset,
+  ColumnCategories,
+  cohortKey,
+  ChartTypes,
+  IGenericChartProps,
+  ISelectorConfig,
+  MissingParametersPlaceholder,
+  defaultModelAssessmentContext,
+  ModelAssessmentContext,
+  rowErrorSize
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import { AccessibleChart } from "@responsible-ai/mlchartlib";
+import _ from "lodash";
+import {
+  getTheme,
+  IDropdownOption,
+  Dropdown,
+  DefaultButton,
+  Text,
+  IChoiceGroupOption
+} from "office-ui-fabric-react";
+import React from "react";
+
+import { defaultTabStyles } from "./DefaultTab.styles";
+import { generatePlotlyProps } from "./generatePlotlyProps";
+import { SidePanel } from "./SidePanel";
+
+export class IDefaultTabProps {}
+
+export interface IDefaultTabState {
+  xDialogOpen: boolean;
+  yDialogOpen: boolean;
+  colorDialogOpen: boolean;
+  selectedCohortIndex: number;
+  chartProps?: IGenericChartProps;
+}
+
+export class DefaultTab extends React.Component<
+  IDefaultTabProps,
+  IDefaultTabState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  private readonly chartAndConfigsId = "DatasetExplorerChart";
+
+  public constructor(props: IDefaultTabProps) {
+    super(props);
+
+    this.state = {
+      colorDialogOpen: false,
+      selectedCohortIndex: 0,
+      xDialogOpen: false,
+      yDialogOpen: false
+    };
+  }
+
+  public componentDidMount(): void {
+    const initialCohortIndex = 0;
+
+    this.setState({
+      chartProps: this.generateDefaultChartAxes(),
+      selectedCohortIndex: initialCohortIndex
+    });
+  }
+
+  public componentDidUpdate(
+    _preProp: IDefaultTabProps,
+    preState: IDefaultTabState
+  ): void {
+    if (preState.selectedCohortIndex >= this.context.errorCohorts.length) {
+      this.setState({ selectedCohortIndex: 0 });
+    }
+  }
+
+  public render(): React.ReactNode {
+    const classNames = defaultTabStyles();
+
+    if (!this.context.jointDataset.hasDataset) {
+      return (
+        <MissingParametersPlaceholder>
+          {localization.Interpret.DatasetExplorer.missingParameters}
+        </MissingParametersPlaceholder>
+      );
+    }
+
+    if (this.state.chartProps === undefined) {
+      return <div />;
+    }
+
+    const selectedCohortIndex =
+      this.state.selectedCohortIndex >= this.context.errorCohorts.length
+        ? 0
+        : this.state.selectedCohortIndex;
+
+    const plotlyProps = generatePlotlyProps(
+      this.context.jointDataset,
+      this.state.chartProps,
+      this.context.errorCohorts.map((errorCohort) => errorCohort.cohort)[
```

Comment:
I think you can remove the map altoghether like this:
```
this.context.errorCohorts[selectedCohortIndex].cohort
```

Reply:
I renamed `DatasetExplorerTab` to `DefaultTab` and Git picked up as a completely new file instead of a file rename. So I did not change any code inside the Dataset Explorer :)